### PR TITLE
packaging/fedora: Merge changes from Fedora Dist-Git

### PR DESCRIPTION
**Pull request by Conan-Kudo**
Fedora Dist-Git carries changes that should be merged back into the snapd reference packaging files.

This also contains the beginnings for Red Hat Enterprise Linux/CentOS 7 support.
However, the packaging is still broken, so it isn't going to be enabled yet.

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@mvo5 This also needs to be merged into 2.29.

File: packaging/fedora/snapd.spec [hunk 1]
```diff
@@ -75,8 +92,11 @@ BuildRequires:  systemd
 
 Requires:       snap-confine%{?_isa} = %{version}-%{release}
 Requires:       squashfs-tools
-# we need squashfs.ko loaded
-Requires:       kmod(squashfs.ko)
+# snapd will use this in the event that squashfs.ko isn't available (cloud instances, containers, etc.)
+# FIXME: Use rich deps for this once Bodhi is switched to using pungi
+Requires:       squashfuse
```
Comment (mvo5) [inline packaging/fedora/snapd.spec]: \o/

File: packaging/fedora/snapd.spec [hunk 2]
```diff
@@ -47,6 +47,23 @@
 
 %global snappy_svcs     snapd.service snapd.socket snapd.autoimport.service snapd.refresh.timer snapd.refresh.service
 
+# Until we have a way to add more extldflags to gobuild macro...
+%if 0%{?fedora} >= 26
+%define gobuild_static(o:) go build -buildmode pie -compiler gc -tags=rpm_crashtraceback -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags -static'" -a -v -x %{?**};
```
Comment (bboozzoo) [inline packaging/fedora/snapd.spec]: Really like enabling `-v -x` 👍 Invaluable when hunting down the 'weird' stuff.

Comment (Conan-Kudo) [reply packaging/fedora/snapd.spec]: This is actually part of the default flags for `%gobuild` in Fedora. :)

Comment (zyga) [reply packaging/fedora/snapd.spec]: o_O not even sure what's going but I'm in awe for that non-trivial line.

File: packaging/fedora/snapd.spec [hunk 3]
```diff
@@ -220,6 +242,7 @@ Provides:      bundled(golang(gopkg.in/yaml.v2))
 %endif
 
 # Generated by gofed
+Provides:      golang(%{import_path}/apparmor) = %{version}-%{release}
```
Comment (zyga) [inline packaging/fedora/snapd.spec]: This directory is no longer in the tree.
If this is stale is the rest correct?

Comment (Conan-Kudo) [reply packaging/fedora/snapd.spec]: I can regenerate the Provides for 2.29.2, I just hadn't done it yet. This is from 2.28.5.

**Pull request closed without merging**